pkg/manifest/a11y.go

```diff
@@ -171,14 +174,70 @@ const (
 	EPUBA11y10WCAG20AA A11yProfile = "http://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-aa"
 	// EPUB Accessibility 1.0 - WCAG 2.0 Level AAA
 	EPUBA11y10WCAG20AAA A11yProfile = "http://www.idpf.org/epub/a11y/accessibility-20170105.html#wcag-aaa"
+	// EPUB Accessibility 1.1 - WCAG 2.0 Level A
+	EPUBA11y11WCAG20A A11yProfile = "https://www.w3.org/TR/epub-a11y-11#wcag-2.0-a"
+	// EPUB Accessibility 1.1 - WCAG 2.0 Level AA
+	EPUBA11y11WCAG20AA A11yProfile = "https://www.w3.org/TR/epub-a11y-11#wcag-2.0-aa"
+	// EPUB Accessibility 1.1 - WCAG 2.0 Level AAA
+	EPUBA11y11WCAG20AAA A11yProfile = "https://www.w3.org/TR/epub-a11y-11#wcag-2.0-aaa"
+	// EPUB Accessibility 1.1 - WCAG 2.1 Level A
+	EPUBA11y11WCAG21A A11yProfile = "https://www.w3.org/TR/epub-a11y-11#wcag-2.1-a"
+	// EPUB Accessibility 1.1 - WCAG 2.1 Level AA
+	EPUBA11y11WCAG21AA A11yProfile = "https://www.w3.org/TR/epub-a11y-11#wcag-2.1-aa"
+	// EPUB Accessibility 1.1 - WCAG 2.1 Level AAA
+	EPUBA11y11WCAG21AAA A11yProfile = "https://www.w3.org/TR/epub-a11y-11#wcag-2.1-aaa"
+	// EPUB Accessibility 1.1 - WCAG 2.2 Level A
+	EPUBA11y11WCAG22A A11yProfile = "https://www.w3.org/TR/epub-a11y-11#wcag-2.2-a"
+	// EPUB Accessibility 1.1 - WCAG 2.2 Level AA
+	EPUBA11y11WCAG22AA A11yProfile = "https://www.w3.org/TR/epub-a11y-11#wcag-2.2-aa"
+	// EPUB Accessibility 1.1 - WCAG 2.2 Level AAA
+	EPUBA11y11WCAG22AAA A11yProfile = "https://www.w3.org/TR/epub-a11y-11#wcag-2.2-aaa"
 )
 
+// Used for sorting. Make sure to keep it up-to-date with the consts
+var a11yProfileRanking = map[A11yProfile]int{
+	EPUBA11y10WCAG20A:   1,
+	EPUBA11y10WCAG20AA:  2,
+	EPUBA11y10WCAG20AAA: 3,
+	EPUBA11y11WCAG20A:   4,
+	EPUBA11y11WCAG20AA:  5,
+	EPUBA11y11WCAG20AAA: 6,
+	EPUBA11y11WCAG21A:   7,
+	EPUBA11y11WCAG21AA:  8,
+	EPUBA11y11WCAG21AAA: 9,
+	EPUBA11y11WCAG22A:   10,
+	EPUBA11y11WCAG22AA:  11,
+	EPUBA11y11WCAG22AAA: 12,
+}
+
 func A11yProfilesFromStrings(strings []string) []A11yProfile {
 	return fromStrings(strings, func(str string) A11yProfile {
 		return A11yProfile(str)
 	})
 }
 
+func (p A11yProfile) Compare(other A11yProfile) int {
+	// Compare based on the compatibility level
+	if p == other {
+		return 0
+	}
+
+	pRank := a11yProfileRanking[p]
+	oRank := a11yProfileRanking[other]
+	return oRank - pRank
+}
+
+type A11yProfileList []A11yProfile
+
+func (l A11yProfileList) Sort() {
```

Comment:
Is this API really useful for integrators, or should we keep it internal for now? It seems we pre-sort the list during parsing.

Reply:
Makes sense to me, I've heard about a few organizations that would like to distribute content that they author as HTML rather than EPUB, so they could be the target for that.